app/controllers/replies_controller.rb

```diff
@@ -166,8 +166,22 @@ def update
     @reply.assign_attributes(reply_params)
     preview(@reply) and return if params[:button_preview]
 
-    @reply.skip_post_update = true unless @reply.post.last_reply_id == @reply.id
-    @reply.save
+    if current_user.id != @reply.user_id && @reply.audit_comment.blank?
+      flash[:error] = "You must provide a reason for your moderator edit."
+      editor_setup
+      render action: :edit and return
+    end
+
+    @reply.audit_comment = nil if @reply.changes.empty? # don't save an audit for a note and no changes
+    @reply.skip_post_update = true unless @reply.post.try(:last_reply_id) == @reply.id
```

Comment:
Also not moderation relevant, but added the missing reply save validation check here, along with a test; since my test checks by setting post to `nil`, also needed a `try()`.